lib/trino-filesystem-cache-alluxio/src/main/java/io/trino/filesystem/alluxio/AlluxioFileSystemCacheConfig.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.filesystem.alluxio;
+
+import com.google.common.base.Splitter;
+import com.google.common.collect.ImmutableList;
+import io.airlift.configuration.Config;
+import io.airlift.configuration.ConfigDescription;
+import io.airlift.configuration.validation.FileExists;
+import io.airlift.units.DataSize;
+import io.airlift.units.Duration;
+import io.airlift.units.MaxDataSize;
+import io.airlift.units.MinDataSize;
+import io.airlift.units.MinDuration;
+import jakarta.validation.constraints.Max;
+import jakarta.validation.constraints.Min;
+import jakarta.validation.constraints.NotNull;
+
+import java.util.List;
+
+import static com.google.common.base.MoreObjects.firstNonNull;
+import static com.google.common.collect.ImmutableList.toImmutableList;
+
+public class AlluxioFileSystemCacheConfig
+{
+    private static final Splitter SPLITTER = Splitter.on(',').trimResults().omitEmptyStrings();
+
+    static final String CACHE_DIRECTORIES = "fs.cache.directories";
+    static final String CACHE_MAX_SIZES = "fs.cache.max-sizes";
+    static final String CACHE_MAX_PERCENTAGES = "fs.cache.max-disk-usage-percentages";
+
+    private List<String> cacheDirectories;
+    private List<DataSize> maxCacheSizes = ImmutableList.of();
+    private Duration cacheTTL = Duration.valueOf("7d");
+    private List<Integer> maxCacheDiskUsagePercentages = ImmutableList.of();
+    private DataSize cachePageSize = DataSize.valueOf("1MB");
+
+    @NotNull
+    public List<@FileExists String> getCacheDirectories()
```

Comment:
Alluxio will create the directory if it does not exist. I've updated the validation mechanism to check if any parent directory is writable.

I've added a product test which tests the configuration and the caching.

I encountered some problems when running the product tests. The Trino nodes launched by the product tests crashed with an error message saying that Java was not installed. The root cause turned out to be that the JDK downloaded by the product test to `/tmp/ptl-tmp-download/temurin21-amd64` was corrupted and missing the library `libjli.so`. Deleting `/tmp/ptl-tmp-download` solved the issue. I'm not exactly sure how this state came about, but thought I'd write it down for posterity.

The error message on `java.nio.file.NoSuchFileException: /opt/data/LOCAL` seems to come from Alluxio's TTL mechanism. From what I can tell, this code will always throw when encountering an empty cache. It's not entirely clear if it's innocent or not, as it might cause the `scheduleAtFixedRate` to stop executing.